### PR TITLE
[Editor] Allow drag and drop of EntityComponent

### PR DESCRIPTION
**Pull request by Eideren**
# PR Details
Also includes a fix to avoid crashing the editor when a file which previously had a script now doesn't anymore but is still drag and dropped onto an entity.

## Related Issue
None

## Types of changes
- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [ ] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: sources/editor/Stride.Assets.Presentation/AssetEditors/EntityComponentSourceCodeResolver.cs [hunk 1]
```diff
@@ -72,35 +72,40 @@ public async Task AnalyzeProject(SessionViewModel session, Project project, Canc
 
             var assemblyFullName = gameProjectCompilation.Assembly.Name;
 
-            var strideScriptType = typeof(ScriptComponent);
+            var strideScriptType = typeof(EntityComponent);
```
Comment (manio143) [inline sources/editor/Stride.Assets.Presentation/AssetEditors/EntityComponentSourceCodeResolver.cs:75]: nit: maybe rename variable to `strideComponentType`?

File: sources/editor/Stride.Assets.Presentation/AssetEditors/EntityHierarchyEditor/ViewModels/AddAssetPolicies/AddEntityComponentFileAssetPolicy.cs [hunk 2]
```diff
@@ -13,15 +13,15 @@
...
+    internal class AddEntityComponentFileAssetPolicy : CreateComponentPolicyBase<ScriptSourceFileAsset, ScriptSourceFileAssetViewModel>
     {
         /// <inheritdoc />
         protected override bool CanAddOrInsert(EntityHierarchyItemViewModel parent, ScriptSourceFileAssetViewModel asset, AddChildModifiers modifiers, int index, out string message, params object[] messageArgs)
         {
             var scriptType = FindScriptType(asset.ServiceProvider, asset.AssetItem)?.FirstOrDefault();
             if (scriptType == null)
             {
-                message = $"No scripts inheriting from {nameof(ScriptComponent)} found in asset {asset.Url}";
+                message = $"No scripts inheriting from {nameof(EntityComponent)} found in asset {asset.Url}";
```
Comment (manio143) [inline sources/editor/Stride.Assets.Presentation/AssetEditors/EntityHierarchyEditor/ViewModels/AddAssetPolicies/AddEntityComponentFileAssetPolicy.cs:24]: nit: "No components inheriting..."

**Review by manio143 — APPROVED** [review]
Looks good. Just two small comments.

**Comment by Eideren** [comment]
Rebased to fix merge issue

**Pull request merged**